pkg/kubelet/kubelet.go

```diff
@@ -666,6 +667,8 @@ type Kubelet struct {
 	reservation kubetypes.Reservation
 }
 
+var _ stats.KubeletStatsProvider = &Kubelet{}
```

Comment:
I ended up removing this.  The way we have our interfaces set up is due to having the Kubelet inject itself into its dependencies without creating a dependency cycle.  I talked with @vishh about this today, and he has some good ideas on how improve this (Dependency Injection).